### PR TITLE
CASMNET-982: Add new pools and peers to the metallb config for Bi-CAN

### PR DESCRIPTION
**Pull request by johren-hpe**
#### Summary and Scope

- Fixes CASMNET-982

##### Issue Type

- RFE Pull Request

This adds the CMN peers and pools to the metallb config.

```
 peers:
    - peer-address: 10.101.8.2
      peer-asn: 65533
      my-asn: 65536
    - peer-address: 10.101.8.3
      peer-asn: 65533
      my-asn: 65536
    - peer-address: 10.252.0.2
      peer-asn: 65533
      my-asn: 65533
    - peer-address: 10.252.0.3
      peer-asn: 65533
      my-asn: 65533
```

```
    address-pools:
    - name: customer-access
      protocol: bgp
      addresses:
      - 10.101.10.128/25
    - name: customer-access-static
      protocol: bgp
      addresses:
      - 10.101.10.112/28
    - name: customer-management
      protocol: bgp
      addresses:
      - 10.101.8.128/25
    - name: customer-management-static
      protocol: bgp
      addresses:
      - 10.101.8.112/28
    - name: hardware-management
      protocol: bgp
      addresses:
      - 10.94.100.0/24
    - name: node-management
      protocol: bgp
      addresses:
      - 10.92.100.0/24
```

#### Prerequisites

- [ x] I have included documentation in my PR (or it is not required) - not required
- [x ] I tested this on internal system (wasp)
    Tested `csi config init` on wasp and verified that the generated metallb.yaml file matches what we have been installing manually.

 
#### Risks and Mitigations
 
No more risk than there was previously for 1.2.   The generated metallb.yaml before this mod was incomplete and had to be fixed manually.

### CODE REVIEW

**Review by jacobsalmela — APPROVED** [review]

File: .version [hunk 1]
```diff
@@ -1 +1 @@
-1.9.7
+1.9.8
```
Comment (rustydb) [inline .version:1]: We published 1.9.8 in 1.2alpha13, can you rebase this and bump to 1.9.9 @johren-hpe ?

File: cmd/init.go [hunk 2]
```diff
@@ -385,6 +385,7 @@ func init() {
 
 	// Use these flags to prepare the basecamp metadata json
 	initCmd.Flags().String("bgp-asn", "65533", "The autonomous system number for BGP conversations")
+	initCmd.Flags().String("bgp-cmn-asn", "65536", "The autonomous system number for CMN BGP clients")
```
Comment (rustydb) [inline cmd/init.go:388]: nit: does 65536 print off in the help or should we add a parenthetical stating the default?

File: pkg/pit/metallb.go [hunk 3]
```diff
@@ -66,9 +73,10 @@ func WriteMetalLBConfigMap(path string, v *viper.Viper, networks map[string]*csi
 	}
 	var configStruct MetalLBConfigMap
 	configStruct.Networks = make(map[string]string)
-	configStruct.ASN = v.GetString("bgp-asn")
```
Comment (rustydb) [inline pkg/pit/metallb.go:76]: Should this be deleted?

**Review by rustydb — CHANGES_REQUESTED** [review]

File: cmd/init.go (hunk 2 above)
Comment (johren-hpe) [reply cmd/init.go:386]: yes, it shows in the help
```
redbull-ncn-m001-pit:~/johren # csi config init --help | grep asn
      --bgp-asn string                        The autonomous system number for BGP conversations (default "65533")
      --bgp-cmn-asn string                    The autonomous system number for CMN BGP clients (default "65536")
      ```

File: .version (hunk 1 above)
Comment (johren-hpe) [reply .version]: Done

File: pkg/pit/metallb.go (hunk 3 above)
Comment (johren-hpe) [reply pkg/pit/metallb.go:76]: yes, that was an variable that used the same ASN for both "peer-asn" and "my-asn" and for all peers.   Now it sets the ASN separately for each peer.

File: pkg/pit/metallb.go [hunk 4]
```diff
@@ -66,9 +73,10 @@ func WriteMetalLBConfigMap(path string, v *viper.Viper, networks map[string]*csi
...
 
 	var spineSwitchXnames, aggSwitchXnames []string
+        var bgpPeers = v.GetString("bgp-peers")
```
[the remark this answers is not in the document]
Comment (johren-hpe) [reply pkg/pit/metallb.go:78]: Fixed

**Review by heemstra — APPROVED** [review]

**Review by rustydb — APPROVED** [review]

**Pull request merged**